File: parametric_tests/anova.r
df <- read.csv("example_data/iris.csv", stringsAsFactors = TRUE)

# Box plots
boxplot(Petal.Width ~ Species, data = df)

# Strip plot
library(lattice)
stripplot(Petal.Width ~ Species, data = df, jitter = TRUE)

# ANOVA
model <- lm(Petal.Width ~ Species, data = df)
summary(aov(model))
# OR
anova(model)

# Tukey HSD
TukeyHSD(aov(model))

# Check assumptions
par(mfrow = c(2, 2))
plot(model) ## QQ plots and others

# Shapiro-Wilk test for normality
shapiro.test(df$Petal.Width)

# Bartlett test for homogenety of variances
bartlett.test(Petal.Width ~ Species, data = df)

# Levene's test for homogeneity of variances
library(car)
leveneTest(Petal.Width ~ Species, data = df)

File: parametric_tests/anova.js
// Load the anova-pairs library.
const anova = require("anova-pairs");

// The data for the ANOVA test.
const data = [
  // Group 1.
  [1, 2, 3],
  // Group 2.
  [2, 3, 4],
  // Group 3.
  [3, 4, 5],
];

// Perform the ANOVA test and calculate the F-value and p-value.
const result = anova.oneway(data);
const fValue = result.fValue;
const pValue = result.pValue;

// Print the F-value and p-value.
console.log(`F-value: ${fValue}`);
console.log(`p-value: ${pValue}`);